File: videosdk-live/videosdk-rtc-android-java-call-trigger-example/call-trigger-app/app/src/main/java/live/videosdk/call_trigger/example/Services/MyInCallService.java
package live.videosdk.call_trigger.example.Services;

import android.content.pm.PackageManager;
import android.telecom.Call;
import android.telecom.InCallService;
import android.telecom.TelecomManager;

import androidx.core.app.ActivityCompat;

public class MyInCallService extends InCallService {

    @Override
    public void onCallAdded(Call call) {
        super.onCallAdded(call);
        call.registerCallback(new Call.Callback() {
            @Override
            public void onStateChanged(Call call, int state) {
                super.onStateChanged(call, state);
                if (state == Call.STATE_ACTIVE) {
                    // Handle the active call state
                }
            }
        });

        // Bring up the default UI for managing the call
        setUpDefaultCallUI(call);
    }

    @Override
    public void onCallRemoved(Call call) {
        super.onCallRemoved(call);
    }

    private void setUpDefaultCallUI(Call call) {
        // Start the default in-call UI
        TelecomManager telecomManager = (TelecomManager) getSystemService(TELECOM_SERVICE);
        if (telecomManager != null) {
            if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
                return;
            }
            telecomManager.showInCallScreen(true);
        }
    }
}

File: videosdk-live/videosdk-rtc-android-kotlin-call-trigger-example/call-trigger-app/app/src/main/java/live/videosdk/call_trigger/kotlin/example/Services/MyInCallService.kt
package live.videosdk.call_trigger.kotlin.example.Services

import android.Manifest
import android.content.pm.PackageManager
import android.telecom.Call
import android.telecom.InCallService
import android.telecom.TelecomManager
import androidx.core.app.ActivityCompat

class MyInCallService : InCallService() {
    override fun onCallAdded(call: Call) {
        super.onCallAdded(call)
        call.registerCallback(object : Call.Callback() {
            override fun onStateChanged(call: Call, state: Int) {
                super.onStateChanged(call, state)
            }
        })
        // Bring up the default UI for managing the call
        setUpDefaultCallUI(call)
    }

    private fun setUpDefaultCallUI(call: Call) {
        // Start the default in-call UI
        val telecomManager = getSystemService(TELECOM_SERVICE) as TelecomManager
        if (telecomManager != null) {
            if (ActivityCompat.checkSelfPermission(
                    this,
                    Manifest.permission.READ_PHONE_STATE
                ) == PackageManager.PERMISSION_GRANTED
            ) {
                telecomManager.showInCallScreen(true)
            }
        }
    }
}
